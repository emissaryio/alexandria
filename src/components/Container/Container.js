import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import classNames from 'classnames';


const styles = theme => ({
  root: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  }
});

const Container = ({
  children,
  classes,
  className,
  ...attributes
}) => (
  <Paper className={classNames(classes.root, className)} elevation={1} {...attributes}>
    {children}
  </Paper>
);

Container.propTypes = {
  square: PropTypes.bool
};

export default withStyles(styles)(Container);
