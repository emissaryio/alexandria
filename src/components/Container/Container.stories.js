import React from 'react';
import { storiesOf } from '@storybook/react';

import { withTests } from '@storybook/addon-jest';
import { MuiThemeProvider } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import { styled } from '@material-ui/styles';
import Theme from '../../hocs/withTheme';
import Container from './Container';


const StyledContainer = styled(Container)({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column'
});

storiesOf('Container', module)
  .addDecorator(withTests('Container'))
  .addWithJSX('default', () => (
    <MuiThemeProvider theme={Theme}>
      <Container className="styled-container">
        <Typography variant="h5" component="h3">
          This is a Component.
        </Typography>
        <Typography component="p">
          This component can be used to build surface or other elements for the application.
        </Typography>
      </Container>
    </MuiThemeProvider>
  ));

storiesOf('Container', module)
  .addDecorator(withTests('Container'))
  .addWithJSX('Styled', () => (
    <MuiThemeProvider theme={Theme}>
      <StyledContainer>
        <Typography variant="h5" component="h3">
          This is a Component.
        </Typography>
        <Typography component="p">
          This component can be used to build surface or other elements for the application.
        </Typography>
      </StyledContainer>
    </MuiThemeProvider>
  ));
