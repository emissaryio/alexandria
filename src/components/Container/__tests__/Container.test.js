import React from 'react';
import { shallow } from 'enzyme';
import Typography from '@material-ui/core/Typography';
import Container from '../Container';

describe('<NavBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Container>
        <Typography variant="h5" component="h3">
          This is a Component.
        </Typography>
        <Typography component="p">
          This component can be used to build surface or other elements for the application.
        </Typography>
      </Container>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
