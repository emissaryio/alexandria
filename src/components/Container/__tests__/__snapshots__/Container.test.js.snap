// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
<Container
  classes={
    Object {
      "root": "Container-root-1",
    }
  }
>
  <WithStyles(Typography)
    component="h3"
    variant="h5"
  >
    This is a Component.
  </WithStyles(Typography)>
  <WithStyles(Typography)
    component="p"
  >
    This component can be used to build surface or other elements for the application.
  </WithStyles(Typography)>
</Container>
`;
